# OCEAN SCIENCES 272:  Problem Set 5

In [1]:
# imports
import numpy as np
from importlib import reload

import unyt

from oceanpy import constants
from oceanpy.forces import coriolis

from os_classes.ocea257 import chapter1, chapter2
from os_classes import plotting as os_plotting

from bokeh import plotting
from bokeh import models

from IPython.display import Image
from IPython.core.display import HTML

# PS5-1-- 3 Waves

## We are to consider a set of 3 waves:

## $\psi_j(x,y,y) = \phi_j(y) \sin[k_j x - \omega t]$

## each growing exponentially with 

## $\phi_j(y,t) = A_j \exp(k_j c_{ij} t)$

## and $A_j = 0.001$ at $t=0$.

## The wavenumbers are given by:

## $\lambda_j = 2\pi/k_j = j * 1000$m

## and the imaginary part is given by:

## $c_{ij} = 100, 300, 350$m/day

In [5]:
# Setup
k_1 = 2 * np.pi / (1000*unyt.m)
k_2 = 2 * np.pi / (2000*unyt.m)
k_3 = 2 * np.pi / (3000*unyt.m)
ks = [k_1, k_2, k_3]

c_1 = 100 * unyt.m/unyt.day
c_2 = 300 * unyt.m/unyt.day
c_3 = 350 * unyt.m/unyt.day
cs = [c_1, c_2, c_3]

## We are asked which wave reaches amplitude 1 first.  This will be the one that maximizes $k_j c_{ij}$

In [10]:
for j, c,k in zip(range(3), cs, ks):
    print("j={},  ck={}".format(j+1, c*k))

j=1,  ck=0.6283185307179586 1/day
j=2,  ck=0.942477796076938 1/day
j=3,  ck=0.7330382858376183 1/day


## It is clear that the $j=2$ wave grows the fastest.

## It will reach $A=1$ at $t = \ln(10^3) / (k_jc_{ij})$ 

In [14]:
t_A2_1 = np.log(1e3) / (k_2 * c_2)
t_A2_1

unyt_quantity(7.32935599, 'day')

## At $t = 7.33$ days, the other two waves have amplitudes:

## $A_j = A(t=0) \exp(k_j c_{ij} t)$

In [15]:
for j, c,k in zip(range(3), cs, ks):
    A_j = 1e-3 * np.exp(c*k*t_A2_1)
    print("j={},  A_j = {}".format(j+1, A_j))

j=1,  A_j = 0.09999999999999996
j=2,  A_j = 0.9999999999999998
j=3,  A_j = 0.2154434690031881


## So $A_1 = 0.1$ and $A_3 = 0.215$

## $A_2 = 10$ occurs $\ln(10) / (k_2 c_{i2})$ days later

In [16]:
t_A2_10 = np.log(1e4) / (k_2 * c_2)
t_A2_10

unyt_quantity(9.77247465, 'day')

In [18]:
for j, c,k in zip(range(3), cs, ks):
    A_j = 1e-3 * np.exp(c*k*t_A2_10)
    print("j={},  A_j = {}".format(j+1, A_j))

j=1,  A_j = 0.46415888336127775
j=2,  A_j = 10.000000000000009
j=3,  A_j = 1.2915496650148832


## Because the waves are all in phase for $x < 1000$m, the superposition is largest in that region.

## At longer wavelengths, the low amplitude, short wavelength $j=1$ wave acts as a 